src/main/java/bgu/spl/mics/application/subscribers/Moneypenny.java

```diff
@@ -10,15 +16,34 @@
  * You MAY change constructor signatures and even add new public constructors.
  */
 public class Moneypenny extends Subscriber {
+	private int _currTime = 0;
+	private Squad squad;
+	private int _serial;
 
-	public Moneypenny() {
-		super("Change_This_Name");
-		// TODO Implement this
+	public Moneypenny(int serial) {
+		super(("Moneypenny" + serial));
+		squad = Squad.getInstance();
+		_serial = serial;
 	}
 
 	@Override
 	protected void initialize() {
-		// TODO Implement this
+		subscribeBroadcast(TickBroadcast.class, (TickBroadcast b)->_currTime = b.getCurrTime());
+		subscribeEvent(AgentsAvailableEvent.class, (AgentsAvailableEvent e)-> {
+			AgentsAvailableObject result = e.getObj();
+			List<String> agentsSerials = result.getAgentsSerials();
+			java.util.Collections.sort(agentsSerials);
+			result.setMoneypennySerial(_serial);
+			if(squad.getAgents(agentsSerials)) {
+				result.setAgentsNames(squad.getAgentsNames(agentsSerials));
+			}
+			complete(e, result);
+			while (!result.isSendMission() || !result.isTerminateMission()) {
+				try {e.wait();} catch (InterruptedException ignored) {}
+			}
```

Comment:
You should probably wait on the result, since M creates the event without saving it